ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/tree/TomlKey.kt

```diff
@@ -38,4 +41,29 @@ public class TomlKey(public val rawContent: String, public val lineNo: Int) {
         }
         return false
     }
+
+    public fun write(emitter: TomlEmitter, config: TomlConfig) {
+        val keys = keyParts
+
+        if (keys.isEmpty() || keys.any(String::isEmpty)) {
```

Comment:
Actually we store `rawContent` in this class already. 
But your logic is more straight-forward, as you work directly with the internal representation of Node. 

What do you think - what is the best thing here to use? `rawContent` or `keyParts`?

Reply:
I see what you're saying, `rawContent.isEmpty()` would work. I think `keyParts` makes more sense though. As you said during encoding the content won't be available unless we format it.

Maybe it'd be better to pass `keyParts` in an internal primary constructor? Something like this perhaps:
```kotlin
public class TomlKey internal constructor(
    internal val keyParts: List<String>,
   public val lineNo: Int
) {
    constructor(rawContent: String, lineNo: Int) : this(rawContent.splitKeyToTokens(lineNo), lineNo)
}
```

For now I was basically working with the existing API with as minimal changes as possible

Reply:
yeah, let's leave that as is and will change while will be making encoding